src/resolve-config.js

```diff
@@ -26,23 +28,47 @@ function getLoadFunction(opts) {
   return getExplorerMemoized(opts).load;
 }
 
-function resolveConfig(filePath, opts) {
+function _resolveConfig(filePath, opts, sync) {
   opts = Object.assign({ useCache: true }, opts);
-  const load = getLoadFunction({ cache: !!opts.useCache, sync: false });
-  return load(filePath, opts.config).then(result => {
-    return !result ? null : mergeOverrides(result, filePath);
-  });
+  const loadOpts = {
+    cache: !!opts.useCache,
+    sync: !!sync,
+    editorconfig: !!opts.editorconfig
+  };
+  const load = getLoadFunction(loadOpts);
+  const loadEditorConfig = resolveEditorConfig.getLoadFunction(loadOpts);
+  const arr = [load, loadEditorConfig].map(l => l(filePath, opts.config));
+
+  const unwrapAndMerge = arr => {
+    const result = arr[0];
+    const editorConfigured = arr[1];
+    const merged = Object.assign(
+      {},
+      editorConfigured,
+      mergeOverrides(Object.assign({}, result), filePath)
+    );
+
+    if (Object.keys(merged).length === 0) {
```

Comment:
Shouldn't this be:

```js
if (!result && !editorConfigured) {
  return null;
}
```

?

Reply:
(e.g. `.prettierrc` file containing `{}` should return `{}`, not `null`)

Reply:
Good catch, fixed in https://github.com/prettier/prettier/pull/3255/commits/e3460b888f1752904c2af1cd50705da3f4ec1a39! This change exposed a bug in some of the tests added in this PR, so I had to remove them. Note that the bug in question would also affect a similar test added on master, so I think this is ok for now. See https://github.com/prettier/prettier/pull/3255/commits/a2c81fc0eea2e9c4b3cee70195d91d01aef0bd3c for details.